src/processors/AvroSchemaInputProcessor.ts

```diff
@@ -0,0 +1,45 @@
+import { AbstractInputProcessor } from './AbstractInputProcessor';
+import { InputMetaModel } from '../models';
+import { Logger } from '../utils';
+import { convertToMetaModel } from '../helpers';
+
+/**
+ * Class for processing Avro Schema input
+ */
+export class AvroSchemaInputProcessor extends AbstractInputProcessor {
+  /**
+   * Function processing an Avro Schema input
+   *
+   * @param input
+   */
+
+  shouldProcess(input?: any): boolean {
+    if (
+      input === '' ||
+      JSON.stringify(input) === '{}' ||
+      JSON.stringify(input) === '[]'
+    ) {
+      return false;
+    }
+    if (!input.type) {
+      return false;
+    }
+    return true;
+  }
+
+  process(input?: any): Promise<InputMetaModel> {
+    if (!this.shouldProcess(input)) {
+      return Promise.reject(
+        new Error('Input is not an Avro Schema, so it cannot be processed.')
+      );
+    }
+    Logger.debug('Processing input as Avro Schema document');
+    const inputModel = new InputMetaModel();
+    inputModel.originalInput = input;
+    const metaModel = convertToMetaModel(input);
```

Comment:
You cannot use this function at all here as it's specific for the use of `CommonModel` types. You have to build your own function that iterates over the input building all relevant Meta Models that the input represents.

Reply:
For example when encountering `record` type, generate a `ObjectMetaModel`, `enum` generate a `EnumMetaModel` 🙂

Reply:
Can you please guide me to test the generated data model from Avro Schema. I tried the same using playground UI in local development but could not obtain the appropriate result since it only works for JSON Schema or AsyncAPI, not sure about it.